selenium-webdriver/index.d.ts

```diff
@@ -3721,17 +3644,63 @@ declare namespace webdriver {
+     * cannot create a session with all of the required capabilities, it will
+     * return an {@linkplain error.SessionNotCreatedError}.
+     *
+     *     let required = new Capabilities().set('browserName', 'firefox');
+     *     let desired = new Capabilities().set('version', '45');
+     *     let driver = WebDriver.createSession(executor, {required, desired});
+     *
+     * This function will always return a WebDriver instance. If there is an error
+     * creating the session, such as the aforementioned SessionNotCreatedError,
+     * the driver will have a rejected {@linkplain #getSession session} promise.
+     * It is recommended that this promise is left _unhandled_ so it will
+     * propagate through the {@linkplain promise.ControlFlow control flow} and
+     * cause subsequent commands to fail.
+     *
+     *     let required = Capabilities.firefox();
+     *     let driver = WebDriver.createSession(executor, {required});
+     *
+     *     // If the createSession operation failed, then this command will also
+     *     // also fail, propagating the creation failure.
+     *     driver.get('http://www.google.com').catch(e => console.log(e));
+     *
      * @param {!command.Executor} executor The executor to create the new session
      *     with.
-     * @param {!./capabilities.Capabilities} desiredCapabilities The desired
+     * @param {(!Capabilities|
+     *          {desired: (Capabilities|undefined),
+     *           required: (Capabilities|undefined)})} capabilities The desired
      *     capabilities for the new session.
      * @param {promise.ControlFlow=} opt_flow The control flow all driver
      *     commands should execute under, including the initial session creation.
      *     Defaults to the {@link promise.controlFlow() currently active}
      *     control flow.
+     * @param {(function(new: WebDriver,
+     *                   !IThenable<!Session>,
+     *                   !command.Executor,
+     *                   promise.ControlFlow=))=} opt_ctor
+     *    A reference to the constructor of the specific type of WebDriver client
+     *    to instantiate. Will create a vanilla {@linkplain WebDriver} instance
+     *    if a constructor is not provided.
+     * @param {(function(this: void): ?)=} opt_onQuit A callback to invoke when
+     *    the newly created session is terminated. This should be used to clean
+     *    up any resources associated with the session.
      * @return {!WebDriver} The driver for the newly created session.
      */
-    static createSession(executor: Executor, desiredCapabilities: Capabilities, opt_flow?: webdriver.promise.ControlFlow): WebDriver;
+    // This method's arguments are untyped so that its overloads can have correct types.
+    // Typescript doesn't allow static methods to be overridden with incompatible signatures.
+    static createSession(...var_args: any[]): WebDriver;
```

Comment:
See the comment on the line above. The child classes of WebDriver have their own createSession methods with incompatible signatures, which typescript forbids. We can either pass this `...var_args: any[]` or define all arguments as type `any` (which may not work anyway, I'd have to check).